tests/test_backend.py

```diff
@@ -202,6 +202,7 @@ def test_expand_dims_more_unknown_rank(self):
             self._test_expand_dims_more_unknown_rank(i)
 
     @check_opset_min_version(9, "ConstantOfShape")
+    @check_opset_after_tf_version("2.2", 12, "ConstantOfShape")
```

Comment:
uh, this is a problem. Why do we need this?

Reply:
basically, onnx has ConstantOfShape that has not changed since opset-9. So why do we depend on opset-12?

Reply:
Because the implementation of MatrixDiag...V3 depends a lot on opset 12 Max/Min which accept integer as input.

Reply:
check_opset_after_tf_version("2.2", 12, "ConstantOfShape") ... the last parameter is the reason. If I read your comment correctly ConstantOfShape is wrong.
But I think this is a something else wrong here:

We handle tf.eye() here:
https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/rewriter/eye_rewriter.py#L26

see the where the fix would be ?

Reply:
Have changed last parameter to "MatrixDiag", does it make sense now?

Reply:
no no.
Look at the re-writer - it is suppose MatrixDiag out and replace it with the onnx EyeLike op.
But it doesn't know about the newer MatrixDiag* versions.

Reply:
I mean why would we go via matrix diag when we have a eye operator.

Reply:
we have fix this in the rewriter because most don't have opset-12 (ie. windows, tensorrt) and for things we supported since ages we should avoid requiring such a new opset.

Reply:
Done, fixed rewriter.